[CxxInterop] use [[msvc::no_unique_address]] on Windows (NFC)

hnrklssn opened the pull request:
This test case uses [[no_unique_address]], which is not supported on Windows. Resolves https://github.com/swiftlang/swift/issues/86095

hnrklssn committed:
[CxxInterop] use [[msvc::no_unique_address]] on Windows (NFC)

This test case used [[no_unique_address]], which is not supported on Windows.
Use [[msvc::no_unique_address]] instead.
Resolves https://github.com/swiftlang/swift/issues/86095

hnrklssn commented:
@swift-ci please smoke test

j-hui reviewed (approved):
looks good to me (LGTM)

hnrklssn committed:
Apply suggestion from @compnerd

Co-authored-by: Saleem Abdulrasool <compnerd@compnerd.org>

hnrklssn committed:
don't share MemberVariables module when testing diagnostics

This header is shared by 3 tests, one of which checks diagnostics. If
that test picks up an already built module from the module cache, it
won't pick up diagnostics in the header. This duplicates the header as
two separate modules rather than specifying an explicit module cache, to
avoid rebuilding CxxStdlib.

hnrklssn commented:
@swift-ci please smoke test
